src/ngResource/resource.js

```diff
@@ -640,7 +642,7 @@ angular.module('ngResource', ['ng']).
         };
 
         forEach(actions, function(action, name) {
-          var hasBody = /^(POST|PUT|PATCH)$/i.test(action.method);
+          var hasBody = /^(POST|PUT|PATCH)$/i.test(action.method) || action.hasBody === true;
```

Comment:
I actually meant it the way I wrote it above. Your version is the same as:

```js
var hasBody = action.hasBody === true || /^(POST|PUT|PATCH)$/i.test(action.method);
```

...which is not what we want.